modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/TransactionsInMultipleThreadsTest.java

```diff
@@ -0,0 +1,423 @@
+import org.apache.ignite.transactions.TransactionIsolation;
+import org.apache.ignite.transactions.TransactionState;
+import org.junit.Assert;
+
+/**
+ *
+ */
+public class TransactionsInMultipleThreadsTest extends AbstractTransactionsInMultipleThreadsTest {
+    /**
+     * @throws Exception  if failed.
+     */
+    public void testSimpleTransactionInAnotherThread() throws IgniteCheckedException {
+        withAllIsolationsAndConcurrencies(new IgniteClosure<Object, Void>() {
+            @Override public Void apply(Object o) {
+                try {
+                    simpleTransactionInAnotherThread();
+                }
+                catch (Exception ignore) {
+                    assert false : "Impossible situation";
+                }
+
+                return null;
+            }
+        });
+    }
+
+    /**
+     * @throws Exception if failed.
+     */
+    private void simpleTransactionInAnotherThread() throws Exception {
+        final IgniteCache<String, Integer> cache = jcache(grid(0), cacheConfiguration(null).setName("testCache"), String.class, Integer.class);
+        final IgniteTransactions transactions = transactions();
+
+        assertNull(cache.get("key1"));
+
+        final Transaction tx = transactions.txStart(transactionConcurrency, transactionIsolation);
+
+        cache.put("key1", 1);
+        cache.put("key2", 2);
+
+        tx.suspend();
+
+        assertNull(cache.get("key1"));
+
+        IgniteInternalFuture<Boolean> fut = GridTestUtils.runAsync(new Callable<Boolean>() {
+            @Override public Boolean call() throws Exception {
+                Assert.assertNull(transactions.tx());
+                Assert.assertEquals(TransactionState.SUSPENDED, tx.state());
+
+                tx.resume();
+
+                Assert.assertEquals(TransactionState.ACTIVE, tx.state());
+
+                cache.put("key3", 3);
+                cache.remove("key2");
+
+                tx.commit();
+
+                return true;
+            }
+        });
+
+        try {
+            fut.get();
+        }
+        catch (IgniteCheckedException e) {
+            e.printStackTrace();
+            fail("Transaction in another thread failed");
+        }
+
+        Assert.assertEquals(TransactionState.COMMITTED, tx.state());
+        Assert.assertEquals((long)1, (long)cache.get("key1"));
+        Assert.assertEquals((long)3, (long)cache.get("key3"));
+        Assert.assertFalse(cache.containsKey("key2"));
+
+        cache.removeAll();
+    }
+
+    /**
+     * @throws IgniteCheckedException
+     */
+    public void testSimpleTransactionInAnotherThreadContinued() throws IgniteCheckedException {
+        withAllIsolationsAndConcurrencies(new IgniteClosure<Object, Void>() {
+            @Override public Void apply(Object o) {
+                try {
+                    simpleTransactionInAnotherThreadContinued();
+                }
+                catch (Exception ignore) {
```

Comment:
Let's rename "e" to "ignored"
Please use fail()
http://junit.sourceforge.net/junit3.8.1/javadoc/junit/framework/Assert.html#fail(java.lang.String)